lib/prepare-driver.js

```diff
@@ -16,53 +16,67 @@ module.exports = function (app){
     var adapterModuleName = adapterDef.moduleName;
     var pathToAdapterDependency = adapterDef.modulePath;
     var adapterConfig = adapterDef.config;
+    var adapterClientName = adapterConfig.adapterClientName || 'redis';
+    var adapterClientConfig = adapterConfig.adapterClientConfig || {};
 
     // For redis:
     // ============================================================
     if (adapterModuleName === 'socket.io-redis') {
 
       // Create raw redis clients if necessary
-      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus;
+      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus || adapterConfig.adapterClientConfig;
       if (!rawClientsNecessary) {
         return cb();
       }
 
-      // Borrow access to the `redis` module from socket.io-redis
+      // Borrow access to the adapter redis client module from socket.io-redis
```

Comment:
Based on the notes above, none of the changes to this try/catch block will be necessary anymore.  Should be able to leave it exactly as it was originally, except wrap the try/catch block in `if/else` statement, where we use the provided redis client, if available-- and otherwise try to require `redis`:

```js
if (!_.isUndefined(redisClient) {
  redis = redisLibrary;
}
else {
  // try/catch block
}
```